controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaDefaultAuthHandler.java

```diff
@@ -0,0 +1,139 @@
+        try (FileReader reader = new FileReader(userPasswdFile);
+             BufferedReader lineReader = new BufferedReader(reader)
+        ) {
+            String line = lineReader.readLine();
+            while ( !Strings.isNullOrEmpty(line)) {
+                if (!line.startsWith("#")) {
+                    String[] userFields = line.split(":");
+                    if (userFields.length >= 2) {
+                        String acls;
+                        if (userFields.length == 2) {
+                            acls = "";
+                        } else {
+                            acls = userFields[2];
+                        }
+                        userMap.put(userFields[0], new PravegaACls(userFields[1], getAcls(acls)));
+                    }
+                }
+                line = lineReader.readLine();
+            }
+        } catch (Exception e) {
+            throw new CompletionException(e);
+        }
+    }
+
+    @Override
+    public String getHandlerName() {
+        return DEFAULT_NAME;
+    }
+
+    @Override
+    public boolean authenticate(Map<String, String> headers) {
+        String userName = headers.get("username");
+        String passwd = headers.get("password");
+        Preconditions.checkArgument(userName != null, "Username not found in header");
+        Preconditions.checkArgument(passwd != null, "Password not found in header");
+
+        StrongPasswordEncryptor encryptor = new StrongPasswordEncryptor();
+        return userMap.containsKey(userName) && encryptor.checkPassword(passwd, userMap.get(userName).encryptedPassword);
+    }
+
+    @Override
+    public PravegaAccessControlEnum authorize(String resource, Map<String, String> headers) {
+        String userName = headers.get("username");
+        if (Strings.isNullOrEmpty(userName) || !userMap.containsKey(userName)) {
+            throw new CompletionException(new PravegaAuthenticationException(userName));
+        }
+        return authorizeForUser(userMap.get(userName), resource);
+
+    }
+
+    @Override
+    public void setServerConfig(Object serverConfig) {
+        loadPasswdFile(((GRPCServerConfig) serverConfig).getUserPasswdFile());
+    }
+
+    private PravegaAccessControlEnum authorizeForUser(PravegaACls pravegaACls, String resource) {
+        PravegaAccessControlEnum retVal = PravegaAccessControlEnum.NONE;
+
+        for (PravegaAcl acl : pravegaACls.acls) {
+            if (acl.resource.equals(resource)) {
+                return acl.acl;
+            } else if (resource.startsWith(acl.resource)) {
+                retVal = acl.acl;
+            } else if (acl.resource.equals("*")) {
+                if (acl.acl.ordinal() > retVal.ordinal()) {
+                    retVal = acl.acl;
+                }
+            }
+        }
+        return retVal;
+    }
+
+    private List<PravegaAcl> getAcls(String aclString) {
+        return  Arrays.stream(aclString.split(";")).map(acl -> {
```

Comment:
Please use `io.pravega.common.util.DelimitedStringParser` if possible. It will eliminate most of your boilerplate code, make the code nicer and catch most syntax problems.

Reply:
You might be able to use above when you parse out the file too.